File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    a = [0] * (n + 1)
    a[0] = 0
    a[1] = 1
    for _ in range(2, n + 1):
        a[_] = a[_ - 2] + a[_ - 1]
    return a[n] % 1234567

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int[] a = new int[n+1];
        a[0] = 0;
        a[1] = 1;
        for(int i=2; i<=n; i++){
            a[i] = (a[i-1] + a[i-2]) % 1234567;  
        }
        answer = a[n];        
        return answer;
    }
}